x-pack/libbeat/cmd/inject.go

```diff
@@ -21,6 +22,10 @@ import (
 	_ "github.com/elastic/beats/v7/x-pack/libbeat/autodiscover/providers/aws/elb"
 )
 
+func init() {
+	version.ElasticLicensed = true
```

Comment:
I would expect this to be retrieved from the Elasticsearch cluster when we check for the version? The same response that is providing the version gives us the Elasticsearch flavour, see `build_flavor` here, which can be `default` or `oss`:

```
{
  "name" : "01e1731f6d10",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "h3NzNl8MSvyhSyNoMGreiw",
  "version" : {
    "number" : "7.10.0-SNAPSHOT",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "6be4acd5094954851a956dd93d8c4043979011d9",
    "build_date" : "2020-11-08T20:00:29.019941Z",
    "build_snapshot" : true,
    "lucene_version" : "8.7.0",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
```

The code that is currently retrieving version info is here:
https://github.com/elastic/beats/blob/d97fe08d6d5bb27e4c5e52d04a45448fd1fe8827/libbeat/esleg/eslegclient/connection.go#L402, check for the `Ping` implementation

Reply:
Ah sorry, I missed the point that this is not needed